docs/map.jinja.rst

```diff
@@ -0,0 +1,197 @@
+.. _map.jinja:
+
+``map.jinja``: gather formula configuration values
+==================================================
+
+The `documentation`_ gives a quick view of the use of a ``map.jinja`` to gather parameters values for a formula.
+
+The today best practice is to let ``map.jinja`` handle parameters from all sources and avoid the call of pillars, grains or configuration from ``sls`` files and template directly.
+
+
+.. contents:: **Table of Contents**
+
+
+How to set configuration values of a formula
+--------------------------------------------
+
+The ``map.jinja`` use several sources where to lookup parameter values:
+
+- YAML files located in the `fileserver`_ under the formula ``parameters/`` directory
+- configuration collected by `salt['config.get']`_
+
+As `pillars`_ are rendered by the SaltStack master for each minion, they could became costly when you have lots of minions with many pillar values.
+
+As a good practice, you should:
+
+- store non secret data in YAML files distributed by the `fileserver`_
+- store sensible data
+
+  - in `pillars`_ (and look for the use of something like `pillar.vault`_)
+  - in `SDB`_ (and look for the use of something like `sbd.vault`_)
```

Comment:
```suggestion
The ``map.jinja`` file uses several sources to lookup parameter values:

- Multiple YAML files organised in the `fileserver`_ under the formula ``parameters/`` directory
- Configuration collected by `salt['config.get']`_

As `pillars`_ are rendered on the Salt master for every minion, this increases the load on the master as the pillar values and the number of minions grow.

As a good practice, you should:

- store non-secret data in YAML files distributed by the `fileserver`_
- store secret data in:

  - `pillars`_ (and look for the use of something like `pillar.vault`_)
  - `SDB`_ (and look for the use of something like `sdb.vault`_)
```

Reply:
Great, much better. And I always do the `sdb` / `sbd` typo :-/